### PR TITLE
Bug 1806914: redirect to topology when resource created from dev-catalog

### PR DESCRIPTION
**Pull request by sahil143**
ODC-Bug: https://issues.redhat.com/browse/ODC-3152

This PR adds functionality to redirect page to the topology page when a resource is installed from dev-catalog.

**Template**
t-to

**Operator-backed**
ob-to

### CODE REVIEW

File: frontend/packages/operator-lifecycle-manager/src/components/create-operand.tsx [hunk 1]
```diff
@@ -426,21 +426,13 @@ export const CreateOperandForm: React.FC<CreateOperandFormProps> = ({
...
-          .then(() =>
-            history.push(
-              `${resourcePathFromModel(
-                ClusterServiceVersionModel,
-                clusterServiceVersion.metadata.name,
-                namespace,
-              )}/${referenceForModel(operandModel)}`,
-            ),
-          )
+          .then(() => history.push(`/topology`))
```
Comment (andrewballantyne) [inline frontend/packages/operator-lifecycle-manager/src/components/create-operand.tsx]: Maybe check for dev perspective? Not sure we want to do this at all times.

File: frontend/packages/operator-lifecycle-manager/src/components/create-operand.tsx [hunk 2]
```diff
@@ -426,15 +426,18 @@ export const CreateOperandForm: React.FC<CreateOperandFormProps> = ({
...
+            const params = new URLSearchParams(window.location.href);
             history.push(
-              `${resourcePathFromModel(
-                ClusterServiceVersionModel,
-                clusterServiceVersion.metadata.name,
-                namespace,
-              )}/${referenceForModel(operandModel)}`,
-            ),
-          )
+              params.has('redirect')
```
Comment (rohitkrai03) [inline frontend/packages/operator-lifecycle-manager/src/components/create-operand.tsx]: Why not use perspective extension to get the redirect URL based on perspective similar to how import from git does it right now?

File: frontend/public/components/catalog/catalog-items.tsx [hunk 3]
```diff
@@ -346,7 +357,7 @@ export class CatalogTileViewPage extends React.Component<
                 <div className="co-catalog-page__overlay-actions">
                   <Link
                     className="pf-c-button pf-m-primary co-catalog-page__overlay-action"
-                    to={detailsItem.href}
+                    to={this.appendTopologyRedirect(detailsItem.href)}
```
Comment (rohitkrai03) [inline frontend/public/components/catalog/catalog-items.tsx]: It's just hard coding the URL redirect to topology which means that user can be in any perspective and still he will be redirected to topology. Is this the behavior that we want here? cc: @christianvogt 

Below screencast shows that even if the user is in admin perspective, he is redirected to topology.
Peek 2020-02-29 01-56

File: frontend/packages/operator-lifecycle-manager/src/components/create-operand.tsx [hunk 4]
```diff
@@ -1115,12 +1118,16 @@ export const CreateOperandYAML: React.FC<CreateOperandYAMLProps> = (props) => {
...
   }
-  const resourceObjPath = () =>
-    `${resourcePathFromModel(
-      ClusterServiceVersionModel,
-      props.match.params.appName,
-      props.match.params.ns,
-    )}/${props.match.params.plural}`;
+  const resourceObjPath = () => {
+    const params = new URLSearchParams(window.location.href);
+    return params.has('redirect')
```
Comment (rohitkrai03) [inline frontend/packages/operator-lifecycle-manager/src/components/create-operand.tsx]: This doesn't work for operator backed services when I tested it. For some reason the `redirect` param that is added is being removed on this page which results in user getting redirected to default admin URL.

Peek 2020-02-29 01-54

Comment (sahil143) [reply frontend/packages/operator-lifecycle-manager/src/components/create-operand.tsx]: @rohitkrai03 Not sure why the param is being removed but I changed the implementation to use activePerspective.

File: frontend/public/components/catalog/catalog-items.tsx (hunk 3 above)
Comment (sahil143) [reply frontend/public/components/catalog/catalog-items.tsx]: This can be avoided now that the implementation is changed to use active Perspective.

**Comment by christianvogt** [comment]
FYI @alecmerdler @spadgett 

We want to provide a catalog extension in the future for contributing items to the dev catalog. In doing that we will refactor how operands are created via the catalog to avoid reusing the same shared route. We will then remove the additions here which redirect to topology.

**Comment by spadgett** [comment]
What happens when the template instance fails?

This might also be really confusing if the template doesn't create any resources that are visible in the topology.

**Comment by spadgett** [comment]
I have a similar question for operator-backed resources. Will the thing you create and its status always be apparent from the topology page?

**Comment by christianvogt** [comment]
@spadgett had the same concerns before and UX said they wanted to always redirect to topology out of the dev catalog.

cc @openshift/team-devconsole-ux

**Comment by serenamarie125** [comment]
We need to handle the error cases appropriately.  What happens now when there's an error @spadgett ?

The topology is the desired location when the user creates/installs something from the dev catalog.  This is the goal for the majority of our use cases.  In edge cases, there may be some oddity, but we should try to understand these cases and see if it's because things should be flagged as hidden from the dev catalog, or what the use case is.

**Comment by andrewballantyne** [comment]
@sahil143 needs a rebase.

**Pull request merged**